app/routers/api/files.py

```
@@ -128,38 +129,146 @@ def _serialize_file(
+    if criteria.tag_id_list:
+        query = query.join(FileInventory.tags).filter(FileTag.tag_id.in_(criteria.tag_id_list)).distinct()
+    if criteria.is_pinned is not None:
+        pinned_subquery = db.query(PinnedFile.file_path)
+        if criteria.is_pinned:
+            query = query.filter(FileInventory.file_path.in_(pinned_subquery))
+        else:
+            query = query.filter(FileInventory.file_path.notin_(pinned_subquery))
+    if criteria.min_size is not None:
+        query = query.filter(FileInventory.file_size >= criteria.min_size)
+    if max_size := criteria.max_size:
+        if max_size >= 0:
```

Ruff (0.15.9)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)